src/modules/luksbootkeyfile/LuksBootKeyFileJob.cpp

```diff
@@ -129,6 +129,31 @@ setupLuks( const LuksDevice& d )
     return true;
 }
 
+// static
+QVariantList
+LuksBootKeyFileJob::partitions()
+{
+    Calamares::GlobalStorage* globalStorage = Calamares::JobQueue::instance()->globalStorage();
+    return globalStorage->value( QStringLiteral( "partitions" ) ).toList();
+}
+
+// static
+bool
+LuksBootKeyFileJob::hasUnencryptedSeparateBoot()
+{
+    const QVariantList partitions = LuksBootKeyFileJob::partitions();
+    for ( const QVariant& partition : partitions )
+    {
+        QVariantMap partitionMap = partition.toMap();
+        QString mountPoint = partitionMap.value( QStringLiteral( "mountPoint" ) ).toString();
+        if ( mountPoint == QStringLiteral( "/boot" ) )
+        {
+            return !partitionMap.contains( QStringLiteral( "luksMapperName" ) );
+        }
+    }
+    return false;
+}
+
```

Comment:
The same applies to this function: it doesn't use the class members, **except** for the `partitions()` function defined above. If you make that function above a free function, then this one can become a free function too.

The advantage here is that you can refactor parts of the program so that it reads like a story, while hiding the implementation details and names from the outside world.